test/functional/Jsr292/src/com/ibm/j9/jsr292/indyn/IndyTest.java

```diff
@@ -370,4 +370,36 @@ public void test_boostrap_return_constant_MethodType() {
 		AssertJUnit.assertEquals(expected, mt);
 		AssertJUnit.assertTrue(expected == mt);
 	}
+
+	// test that if resolved CallSite is null, the same error is rethrown
+	@Test(groups = { "level.extended" })
+	public void test_CallSiteNullErrorRethrown () {
+		// BSME is expected on the first run
+		boolean bootstrapMethodErrorThrown1 = false;
+		try {
+			com.ibm.j9.jsr292.indyn.GenIndyn.test_CallSiteNullErrorRethrown();
+		} catch ( java.lang.BootstrapMethodError e ) {
+			bootstrapMethodErrorThrown1 = true;
+		}
+
+		if ( bootstrapMethodErrorThrown1 == false ) {
+			Assert.fail( "BootstrapMethodError was expected for null CallSite" );
+		}
+
+		// The same BSME is expected on the second run
+		boolean bootstrapMethodErrorThrown2 = false;
+		try {
+			com.ibm.j9.jsr292.indyn.GenIndyn.test_CallSiteNullErrorRethrown();
+		} catch ( java.lang.BootstrapMethodError e ) {
+			if ( "1.8".equals(System.getProperty("java.specification.version")) ) {
```

Comment:
Use https://github.com/eclipse/openj9/blob/058b1e3060558afc1fc90a171c7e5396598921e8/test/functional/TestUtilities/src/org/openj9/test/util/VersionCheck.java#L31 to check the version.